csharp/Adapter/Microsoft.Spark.CSharp/Sql/DataFrame.cs

```diff
@@ -26,14 +24,14 @@ public class DataFrame
         private readonly SparkContext sparkContext;
         [NonSerialized]
         private StructType schema;
-        private RowSchema rowSchema;
         [NonSerialized]
         private RDD<Row> rdd;
-
+        [NonSerialized]
+        private IRDDProxy rddProxy;
         [NonSerialized]
         private bool? isLocal;
         [NonSerialized]
-        private Random random = new Random();
+        private readonly Random random = new Random();
```

Comment:
Our usage of random is pretty trivial and just in one non-critical method which is not multi-threaded. I believe we do not need SafeRandom right away but thanks for the info and we should keep an eye on the bugs in Sample method to see if it ever starts exhibiting the "feature" you described.